README.md

```diff
@@ -9,6 +9,27 @@ This code requires a local copy of JerryScript and Zephyr OS source.  We
 will upstream patches to those projects as appropriate, but this repo is for
 everything else.
 
+### Index
```

Comment:
Yeah I didn't want to open up that can of worms. Really, I started by just adding the QEMU docs, but then realized how terrible and long the readme was. I can though, we just may want to talk about what our main bullets should be, and then sort the existing sections into those categories.